src/hypothesis/strategies.py

```diff
@@ -836,31 +836,80 @@ def dm_func(denom):
 
 @cacheable
 @defines_strategy
-def decimals(
-    min_value=None, max_value=None, allow_nan=None, allow_infinity=None
-):
-    """Generates instances of decimals.Decimal.
+def decimals(min_value=None, max_value=None,
+             allow_nan=True, allow_infinity=None, places=None):
+    """Generates instances of decimals.Decimal, which may be:
 
-    - If min_value is not None, all values will be >= min_value.
-    - If max_value is not None, all values will be <= max_value.
-    - If min_value or max_value is not None, it is an error to enable
-      allow_nan.
-    - If both min_value and max_value are not None, it is an error to enable
-      allow_infinity.
+    - A finite rational number, between ``min_value`` and ``max_value``.
+    - Not a Number, if ``allow_nan`` is truthy.
```

Comment:
I think we've talked about this before, but can you remind me why the behaviour for allow\_nan and bounds is different for decimals than it is for floats? It still seems wrong to me that you can return NaN if you have a bound, because it breaks the invariant that the generated value is in those bounds, but I remember there being some sort of complication.